Libraries/LibWeb/HTML/Dates.cpp

```diff
@@ -243,4 +244,68 @@ WebIDL::ExceptionOr<GC::Ref<JS::Date>> parse_time_string(JS::Realm& realm, Strin
     return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, "Can't parse time string"sv };
 }
 
+// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#parse-a-month-component
+static Optional<YearAndMonth> parse_a_month_component(GenericLexer& input)
+{
+    // 1. Collect a sequence of code points that are ASCII digits from input given position. If the collected sequence is
+    //    not at least four characters long, then fail. Otherwise, interpret the resulting sequence as a base-ten integer.
+    //    Let that number be the year.
+    auto year_string = input.consume_while(is_ascii_digit);
+    if (year_string.length() < 4)
+        return {};
+    auto maybe_year = year_string.to_number<u32>();
+    if (!maybe_year.has_value())
+        return {};
+    auto year = maybe_year.value();
+
+    // 2. If year is not a number greater than zero, then fail.
+    if (year < 1)
+        return {};
+
+    // 3. If position is beyond the end of input or if the character at position is not a U+002D HYPHEN-MINUS character, then
+    //    fail. Otherwise, move position forwards one character.
+    if (!input.consume_specific('-'))
+        return {};
+
+    // 4. Collect a sequence of code points that are ASCII digits from input given position. If the collected sequence is not
+    //    exactly two characters long, then fail. Otherwise, interpret the resulting sequence as a base-ten integer. Let that
+    //    number be the month.
+    auto month_string = input.consume_while(is_ascii_digit);
+    if (month_string.length() != 2)
+        return {};
+    auto month = month_string.to_number<u32>().value();
+
+    // 5. If month is not a number in the range 1 ≤ month ≤ 12, then fail.
+    if (month < 1 || month > 12)
+        return {};
+
+    // 6. Return year and month.
+    return YearAndMonth { year, month };
+}
+
+// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#parse-a-month-string
+Optional<YearAndMonth> parse_a_month_string(StringView input_view)
+{
+    // 1. Let input be the string being parsed.
+    // 2. Let position be a pointer into input, initially pointing at the start of the string.
+    GenericLexer input { input_view };
+
+    // 3. Parse a month component to obtain year and month. If this returns nothing, then fail.
+    auto year_and_month = parse_a_month_component(input);
+    if (!year_and_month.has_value())
+        return {};
+
+    // 4. If position is not beyond the end of input, then fail.
+    if (!input.is_eof())
+        return {};
+
+    // 5. Return year and month.
+    return year_and_month;
+}
+
+u32 number_of_months_since_unix_epoch(YearAndMonth year_and_month)
+{
+    return (year_and_month.year - 1970) * 12 + year_and_month.month - 1;
```

Comment:
What happens if the year is 1969? Seems like this should return a signed result?

Reply:
oops, yeah this is wrong, I'll submit a MR to WPT to add a test for this case. It's also incorrect on the setter side to use fmod as that also doesn't handle the signed case, should instead be something like JS::modulo

Reply:
okay I fixed that. I have a MR open for WPT which includes a test for this case. It's approved but not yet merged. Have speculatively included those tests in the WPT import.